snap7/util.py

```diff
@@ -1175,6 +1175,35 @@ def get_date(bytearray_: bytearray, byte_index: int = 0) -> date:
         raise ValueError("date_val is higher than specification allows.")
     return date_val
 
+def set_date(bytearray_: bytearray, byte_index: int, date: date) -> bytearray:
+    """Set value in bytearray to date
+
+        Notes:
+            Datatype `date` consists in the number of days elapsed from 1990-01-01.
+            It is stored as an int (2 bytes) in the PLC.
+
+        Args:
+            bytearray_: buffer to write.
+            byte_index: byte index from where to start writing.
+            date: date object
+
+        Examples:
+            >>> data = bytearray(2)
+
+            >>> snap7.util.set_date(data, 0, date(2024, 3, 27))
+
+            >>> data
+                bytearray(b'\x30\xd8')
+        """
+    if date < date(1990, 1, 1):
```

Comment:
Your local variable `date` overshadow the function `date`. Implementing a small test could help avoid this oversight